amqp/src/main/scala/akka/stream/alpakka/amqp/scaladsl/AmqpFlow.scala

```diff
@@ -0,0 +1,102 @@
+   *
+   * This stage materializes to a `Future` of `Done`, which can be used to know when the Flow completes,
+   * either normally or because of an amqp failure.
+   */
+  def withConfirm[T](
+      settings: AmqpWriteSettings,
+      confirmationTimeout: FiniteDuration
+  ): Flow[WriteMessage, WriteResult, Future[Done]] =
+    asFlowWithoutContext(
+      Flow.fromGraph(new impl.AmqpBlockingFlowStage(settings, confirmationTimeout))
+    )
+
+  /**
+   * Creates an `AmqpFlow` that accepts `WriteMessage` elements and emits `WriteResult`.
+   *
+   * This variant of `AmqpFlow` asynchronously waits for message confirmations. Maximum number of messages
+   * simultaneously waiting for confirmation before signaling backpressure is configured with a
+   * `bufferSize` parameter. Emitted results preserve the order of messages pulled from upstream - due to that
+   * restriction this flow is expected to be slightly less effective than it's unordered counterpart.
+   *
+   * In case of upstream failure/finish this stage attempts to process all buffered messages (waiting for
+   * confirmation) before propagating failure/finish downstream.
+   *
+   * This stage materializes to a `Future` of `Done`, which can be used to know when the Flow completes,
+   * either normally or because of an amqp failure.
+   */
+  def withAsyncConfirm(
+      settings: AmqpWriteSettings,
+      bufferSize: Int,
+      confirmationTimeout: FiniteDuration
+  ): Flow[WriteMessage, WriteResult, Future[Done]] =
+    asFlowWithoutContext(
+      Flow.fromGraph(new impl.AmqpAsyncFlowStage(settings, bufferSize, confirmationTimeout))
+    )
+
+  /**
+   * Creates an `AmqpFlow` that accepts `WriteMessage` elements and emits `WriteResult`.
+   *
+   * This variant of `AmqpFlow` asynchronously waits for message confirmations. Maximum number of messages
+   * simultaneously waiting for confirmation before signaling backpressure is configured with a
+   * `bufferSize` parameter. Results are emitted downstream as soon as confirmation is received, meaning that
+   * there is no ordering guarantee of any sort.
+   *
+   * In case of upstream failure/finish this stage attempts to process all buffered messages (waiting for
+   * confirmation) before propagating failure/finish downstream.
+   *
+   * This stage materializes to a `Future` of `Done`, which can be used to know when the Flow completes,
+   * either normally or because of an amqp failure.
+   */
+  def withAsyncUnorderedConfirm[T](
+      settings: AmqpWriteSettings,
+      bufferSize: Int,
+      confirmationTimeout: FiniteDuration
+  ): Flow[WriteMessage, WriteResult, Future[Done]] =
+    asFlowWithoutContext(
+      Flow.fromGraph(new impl.AmqpAsyncUnorderedFlowStage(settings, bufferSize, confirmationTimeout))
+    )
+
+  private def asFlowWithoutContext(flow: Flow[(WriteMessage, NotUsed), (WriteResult, NotUsed), Future[Done]]) =
```

Comment:
I think you are the first to turn this around so that `withContext` is the default thing to do. Great!